File: 49 Group Anagrams/soln.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = {}
        
        for s in strs:
            srt = ''.join(sorted(s))
            if srt not in d:
                d[srt] = []
            d[srt].append(s)
        
        return [d[k] for k in d]

File: 49 Group Anagrams/soln.go
func groupAnagrams(strs []string) [][]string {
    m := make(map[string][]string)
    var res [][]string
    
    for i := 0; i < len(strs); i++{
        s := sorted(strs[i])
        if _, ok := m[s]; ok{
            m[s] = append(m[s], string(strs[i]))
        } else{
            m[s] = []string{strs[i]}
        }
    }
    
    for key, _ := range m{
        res = append(res, m[key])
    }
    return res
}

func sorted(str string) string {
    s := strings.Split(str, "")
    sort.Strings(s)
    return strings.Join(s, "")
}